plenario/api/condition_builder.py

```diff
@@ -20,80 +21,78 @@
 }
 
 
-def parse_general(table, field, value):
-    """To parse the non-tree style way of providing arguments.
+def parse_tree(table, condition_tree, literally=False):
```

Comment:
Replacing parse_general is parse_tree, which takes a dictionary of the form {"op": "<sqlalchemy_op>", "col": "<dataset_col>", "val": "<target_value>"} and builds a single, cumulative SQLAlchemy condition from it. Because of the recursive nature of the method, it allows one to nest conditions to express more complicated queries such as (FOO1 = BAR1 AND FOO2 = BAR2) OR (FOO3 >= BAR3)